adoc/chapters/programming_interface.adoc

```diff
@@ -19423,8 +19423,13 @@ these actual types:
 ----
 float fmax(float x, float y)
 double fmax(double x, double y)
-template<int N> vec<float,N> fmax(vec<float,N> x, vec<float,N> y)
-template<size_t M> marray<float,M> fmax(marray<float,M> x, marray<float,M> y)
+vec<float,1> fmax(vec<float,1> x, vec<float,1> y)
+vec<float,2> fmax(vec<float,2> x, vec<float,2> y)
+vec<float,3> fmax(vec<float,3> x, vec<float,3> y)
+vec<float,4> fmax(vec<float,4> x, vec<float,4> y)
+vec<float,8> fmax(vec<float,8> x, vec<float,8> y)
+vec<float,16> fmax(vec<float,16> x, vec<float,16> y)
+template<size_t N> marray<float,N> fmax(marray<float,N> x, marray<float,N> y)
```

Comment:
I guess it is a discrepancy compared to `vec` to solve with concepts in SYCL Next...

Reply:
Our thinking is that `vec` will probably go away at some point.  Once that happens, all of these overloads on `vec` can be removed.  In the meantime, most people wanted to do whatever was easiest for implementors w.r.t. `vec`, and the consensus from implementors was that adding these overloads is the easiest solution.

Reply:
The problem is nit the `vec` but that the `marray` with this template is not user-friendly without deeper rework.